internal/cli/pipeline_list.go

```diff
@@ -86,11 +88,58 @@ func (c *PipelineListCommand) Run(args []string) int {
 
 		totalSteps := strconv.Itoa(len(pipeline.Steps))
 
+		pipelineRunsResp, err := c.project.Client().ListPipelineRuns(c.Ctx, &pb.ListPipelineRunsRequest{
+			Pipeline: &pb.Ref_Pipeline{
+				Ref: &pb.Ref_Pipeline_Id{
+					Id: &pb.Ref_PipelineId{
+						Id: pipeline.Id,
+					},
+				},
+			},
+		})
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+
+		// Note(xx): This will be refactored in a future PR to use a pipeline bundle
+		// that caches this information without having to make a call to the pipeline runs endpoint
+		// for every pipeline and jobs for every pipeline run.
+		runs := pipelineRunsResp.PipelineRuns
```

Comment:
We should be sure to check in the case where we have no runs (i.e. new project or never before run pipelines) otherwise this will panic!

```
brian@ubuntu:waypoint-tetris λ waypoint pipeline list                                                                                                                                                                                                                ±[main]

» Waypoint Pipelines for k8s-tetris
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/hashicorp/waypoint/internal/cli.(*PipelineListCommand).Run(0x4000bc6f90, {0x40002a6930?, 0x16?, 0xffffffffffffffff?})
        /media/psf/go/src/github.com/hashicorp/waypoint/internal/cli/pipeline_list.go:108 +0xc8c
github.com/mitchellh/cli.(*CLI).Run(0x4000b39e00)
        /media/psf/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x4cc
github.com/hashicorp/waypoint/internal/cli.Main({0x400006e090?, 0x46190?, 0x25d0d20?})
        /media/psf/go/src/github.com/hashicorp/waypoint/internal/cli/main.go:127 +0x528
main.main()
        /media/psf/go/src/github.com/hashicorp/waypoint/cmd/waypoint/main.go:14 +0xb4
```

Reply:
There might be something else at play actually...even after a few runs I get this panic.

Reply:
Oh that's really weird, I haven't been able to reproduce this.